omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,145 @@
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value = self.convertDate1($scope.condition.onSetDate);
+          
+            if($scope.condition.status=='INACTIVE'){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = self.convertDate1($scope.condition.endDate);
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[1].value = self.convertDate2($scope.condition.onSetDate);  
+            self.showEndDate();
+        }
+
+    self.showEndDate = self.showEndDate || function() {
+        var inputs = document.getElementsByTagName('input');
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status=='INACTIVE'){
+            groups[2].style.visibility = "visible";
+            inputs[3].value = self.convertDate2($scope.condition.endDate);
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+    }
+
+    self.validateCondition = self.validateCondition || function () {
+            var concept ;
+            if($scope.concept.concept.uuid)
+            {
+                concept=$scope.concept.concept;
+            }
+            else{
+                concept=$scope.condition.concept;
+            }
+            if (concept !== null) {
+                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+            } else {
+                // noncoded diagnosis
+                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+            }
+
+            $scope.condition.onSetDate = self.getSelectedDate();
+            if($scope.condition.status=='INACTIVE')
+            {
+                $scope.condition.endDate = self.getEndDate();
+            }
+            self.saveCondition();
+        }
+
+    self.unselectStatus = self.unselectStatus || function () {
+            $scope.condition.status = null;
+        }
+
+    self.getSelectedDate = self.getSelectedDate || function () {
+            var datePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];
+            return datePicker.value;
+        }
+        
+    self.getEndDate = self.getEndDate || function() {
+           var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];
+           return endDatePicker.value;
+    }    
+
+    self.convertDate1 = self.convertDate1 || function(date_in_ms) {
+            var date = new Date(date_in_ms);
+            var y = date.getFullYear();
+            var m = date.getMonth() + 1;    
+            var d = date.getDate();    
+            m = (m < 10) ? '0' + m : m; 
```

Comment:
What is this code accomplishing?

Reply:
For setting the values of date in the concept.

Reply:
Is that obvious from the code?

Reply:
I will add a comment before the function

Reply:
With time, comments tend to get out of sync with the code. Could you just create a method with the appropriate name?

Reply:
okay i will make the changes

Reply:
I updated the function name

Reply:
What is the function name?

Reply:
convertForDateValue-   For setting the values of date in the concept.
convertForDateDisplay - to display the dates in the onset date and end date fields in the following format- 18 Jun 2019

Reply:
What does convertForDateValue mean?

Reply:
Do you want me to name it to something else? If yes, Please suggest some name to this function.

Reply:
Have you looked at what the rest of the openmrs pages use for this?

Reply:
I did, but I didn't find any implementation that could be of help in this context.

Reply:
How do those other pages display and convert their dates?

Reply:
datepicker in ui commons is using an inbuilt function. If I am allowed to change the format in which date in edit conditions can be displayed, I will use some inbuilt functions.

Reply:
yes use that inbuilt function

Reply:
I updated the PR